requests_toolbelt/adapters/appengine.py

```diff
@@ -44,6 +49,57 @@ def __init__(self, validate_certificate=True, *args, **kwargs):
         super(AppEngineAdapter, self).__init__(self, *args, **kwargs)
 
     def init_poolmanager(self, connections, maxsize, block=False):
-        self.poolmanager = gaecontrib.AppEngineManager(
-            validate_certificate=self._validate_certificate
-        )
+        self.poolmanager = _AEPoolManager(self._validate_certificate)
+
+
+class _AEPoolManager(object):
+    """Implements urllib3's PoolManager API."""
+
+    def __init__(self, validate_certificate=True):
+        self.validate_certificate = validate_certificate
+
+    def connection_from_url(self, url):
+        return _AEConnection(self, url)
```

Comment:
Not sure, it depends on how requests treats connection objects. App Engine has a relatively low memory limit (128mb) so it's better to avoid unnecessary allocations, but it's also not good to accumulate a ton of objects in a indeterminate pool.

I think creating a connection per requests is fine, but please re-use the underlying manager.